tests/upgrades/test_discovery.py

```diff
@@ -0,0 +1,79 @@
+"""Test Discovery Plugin related Upgrade Scenario's
+
+:Requirement: UpgradedSatellite
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: DiscoveryImage
+
+:Team: Rocket
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import re
+
+from packaging import version
+import pytest
+
+
+class TestDiscoveryImage:
+    """Pre-upgrade and post-upgrade scenarios to test Foreman Discovery Image version.
+
+    Test Steps:
+        1. Before Satellite upgrade, Check the FDI version on the Discovered host
+        2. Upgrade satellite.
+        3. Check the FDI version on the Discovered host, after upgrade to make sure its same or greater.
+    """
+
+    @pytest.mark.pre_upgrade
+    def test_pre_upgrade_fdi_version(self, target_sat, save_test_data):
+        """Test FDI version before upgrade.
+
+        :id: preupgrade-8c94841c-6791-4af0-aa9c-e54c8d8b9a92
+
+        :steps:
+            1. Check installed version of FDI
+
+        :expectedresults: Version should be saved and checked post-upgrade
+        """
+        target_sat.register_to_cdn()
+        target_sat.execute(
+            'yum -y --disableplugin=foreman-protector install foreman-discovery-image'
+        )
+        target_sat.unregister()
+        fdi_version = target_sat.execute('rpm -qa *foreman-discovery-image*')
```

Comment:
See if `target_sat.execute(f'rpm -q --qf "%{{VERSION}}" {rpm_name}').stdout` works.

Reply:
Firstly, I think it should be "%{VERSION}". Secondly, I think an f-string will have hard time with the curly braces :)

Reply:
But yes, try something along the lines of what Jameer suggested. The way you're doing it looks too complicated.

Reply:
Thanks for the input I will try it out.

Reply:
I tried out with the release  as well what I am looking for is the whole version '4.1.0-10' 
```
result = target_sat.execute(f'rpm -q --qf "%{{VERSION}}-%{{RELEASE}}" {"foreman-discovery-image"}').stdout
ipdb> result
'4.1.0-10.el8sat'

```
And again I have split and do some modifications here to extract that. So keeping it as it is.